libs/langchain/langchain/llms/llamacpp.py

```diff
@@ -302,6 +314,31 @@ def _call(
             result = self.client(prompt=prompt, **params)
             return result["choices"][0]["text"]
 
+    async def _acall(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> str:
+        """Async call the Llama model and return the output.
+        Args:
+            prompt: The prompt to use for generation.
+            stop: A list of strings to stop generation when encountered.
+        Returns:
+            The generated text.
+        """
+        if self.streaming:
+            combined_text_output = ""
+            async for chunk in self._astream(prompt, stop, run_manager, **kwargs):
+                combined_text_output += chunk.text
+            return combined_text_output
+        else:
+            params = self._get_parameters(stop)
+            params = {**params, **kwargs}
+            result = self.client(prompt=prompt, **params)
```

Comment:
This is incorrect, see self.client needs to be called with `run_in_executor` see comment below for example